src/lib/fixtures/files/ts/Dockerfile

```diff
@@ -1,21 +1,29 @@
 FROM node:18.16.1-alpine
+# Use an official Node.js runtime as a parent image 
 
-RUN apk add --no-cache bash
-RUN npm i -g @nestjs/cli @samagra-x/stencil-cli typescript ts-node
+# Set the working directory in the container
+WORKDIR /usr/src/app
 
-COPY package*.json /tmp/app/
-RUN cd /tmp/app && npm install
+# Copy the package.json and package-lock.json files to the container
+COPY package*.json ./
 
-COPY . /usr/src/app
-RUN cp -a /tmp/app/node_modules /usr/src/app
-COPY ./wait-for-it.sh /opt/wait-for-it.sh
-COPY ./startup.dev.sh /opt/startup.dev.sh
-RUN sed -i 's/\r//g' /opt/wait-for-it.sh
-RUN sed -i 's/\r//g' /opt/startup.dev.sh
+RUN npm i -g @samagra-x/stencil-cli typescript ts-node
+
+# Install dependencies
+RUN npm install
+
+# Copy the rest of the application code to the container
+COPY . .
+
+# Build the NestJS application
 
-WORKDIR /usr/src/app
-RUN cp env-example .env
-RUN npx prisma generate 
 RUN npm run build
 
-CMD ["/opt/startup.dev.sh"]
+# Set environment variable (default value, can be overridden by .env)
+ENV PORT 3000
+ENV ENVIRONMENT DEVELOPMENT
+
+# Expose the port the app runs on
+EXPOSE ${PORT}
+
+CMD ["sh", "-c", "if [ \"$ENVIRONMENT\" = \"PRODUCTION\" ]; then npm run start:prod; else npm run start:dev; fi"]
```

Comment:
I think docker has native support for env vars via the projects root level `.env` files

Reply:
yes it has native support, we are setting default to 3000, can be overridden by `.env` file